prog/vm/nexus.rb

```diff
@@ -359,18 +310,19 @@ def destroy
     end
 
     DB.transaction do
-      vm.private_subnets_dataset.delete
+      vm.nics.map do |nic|
+        nic.update(vm_id: nil)
```

Comment:
When an user created an VM and delete it without knowing anything about networking, he will have leftover nic and subnet.